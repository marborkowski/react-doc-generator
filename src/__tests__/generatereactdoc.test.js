import generateReactDocs from "../generatereactdoc";
import path from "path";

describe("GenerateReactDoc", () => {
  it("Whole thing should work", async () => {
    const output = await generateReactDocs({
      sourceDir: path.resolve(__dirname, "../__mocks__"),
      extensions: ["js", "jsx"],
      outputDir: path.resolve(__dirname, "../__mocks__/output.md"),
    });
    expect(output).toMatchSnapshot();
  });
});
