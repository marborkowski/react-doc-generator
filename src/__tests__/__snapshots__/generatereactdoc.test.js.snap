// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateReactDoc Whole thing should work 1`] = `
Array [
  Object {
    "files": Array [
      Object {
        "components": Array [
          Object {
            "description": "fdgdfgdf gfdgfdg fdgfdgdfg
   
gdfgfdgdfg dfgdfgfdg dfg df getDefaultPropsg fdgfd
   
gfdgfdgdfgfdg.
   
   
   
Example:
   
\`\`\`html
   
<Component name=\\"bebe\\" />
   
\`\`\`   

",
            "methods": Array [],
            "props": Object {
              "bar": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "21",
                },
                "description": "Description of prop \\"bar\\" (a custom validation function).",
                "required": false,
                "type": Object {
                  "name": "custom",
                  "raw": "function(props, propName, componentName) {
  // ...
}",
                },
              },
              "baz": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "",
                "required": false,
                "type": Object {
                  "name": "union",
                  "value": Array [
                    Object {
                      "name": "number",
                    },
                    Object {
                      "name": "string",
                    },
                  ],
                },
              },
              "finger": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "'medium'",
                },
                "description": "Description of prop \\"finger\\".",
                "required": false,
                "type": Object {
                  "name": "string",
                },
              },
              "foo": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "42",
                },
                "description": "Description of prop \\"foo\\".",
                "required": false,
                "type": Object {
                  "name": "number",
                },
              },
              "toe": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "''",
                },
                "description": "Description of prop \\"toe\\" has one break line here following more comments and has default empty string.",
                "required": false,
                "type": Object {
                  "name": "string",
                },
              },
            },
            "title": "General component description.",
          },
        ],
        "filename": "C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\ExampleComponent.js",
      },
      Object {
        "components": Array [
          Object {
            "description": "null   

",
            "displayName": "Component1",
            "methods": Array [],
            "props": Object {
              "bar": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "Description of prop \\"bar\\" (a custom validation function).",
                "required": false,
                "type": Object {
                  "name": "custom",
                  "raw": "function(props, propName, componentName) {
  // ...
}",
                },
              },
              "baz": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "",
                "required": false,
                "type": Object {
                  "name": "union",
                  "value": Array [
                    Object {
                      "name": "number",
                    },
                    Object {
                      "name": "string",
                    },
                  ],
                },
              },
              "foo": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "10000099999",
                },
                "description": "Description of prop \\"foo\\".",
                "required": false,
                "type": Object {
                  "name": "number",
                },
              },
              "onExit": Object {
                "defaultValue": Object {
                  "computed": false,
                  "value": "See code",
                },
                "description": "",
                "required": false,
                "type": Object {
                  "name": "func",
                },
              },
            },
            "title": "Component1",
          },
          Object {
            "description": "Blah blah blah...
   
fdfdfsdf
   
fdsfsd   

",
            "displayName": "DUPA",
            "methods": Array [],
            "props": Object {
              "bar": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "Description of prop \\"bar\\" (a custom validation function).",
                "required": false,
                "type": Object {
                  "name": "custom",
                  "raw": "function(props, propName, componentName) {
  // ...
}",
                },
              },
              "baz": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "",
                "required": false,
                "type": Object {
                  "name": "union",
                  "value": Array [
                    Object {
                      "name": "number",
                    },
                    Object {
                      "name": "string",
                    },
                  ],
                },
              },
              "foo": Object {
                "defaultValue": Object {
                  "value": undefined,
                },
                "description": "Description of prop \\"foo\\".",
                "required": false,
                "type": Object {
                  "name": "number",
                },
              },
            },
            "title": "DUPA",
          },
        ],
        "filename": "C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\inside\\\\OtherExampleComponent.js",
      },
    ],
    "version": "1.2.5",
  },
  Array [
    Array [
      "C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\ExampleComponent.js",
      1,
      "[32mOK.[39m",
    ],
    Array [
      "C:\\\\Users\\\\rravindran\\\\metocean\\\\react-doc-generator\\\\src\\\\__mocks__\\\\inside\\\\OtherExampleComponent.js",
      2,
      "[32mOK.[39m",
    ],
  ],
]
`;
